src/Sulu/Bundle/SnippetBundle/EventListener/DomainEventSubscriber.php

```diff
@@ -0,0 +1,348 @@
+
+        $this->domainEventCollector->collect(
+            new SnippetRemovedEvent(
+                $document->getUuid(),
+                $document->getTitle(),
+                $document->getLocale()
+            )
+        );
+    }
+
+    public function handleRemoveLocale(RemoveLocaleEvent $event): void
+    {
+        $document = $event->getDocument();
+        $locale = $event->getLocale();
+
+        if (!$document instanceof SnippetDocument) {
+            return;
+        }
+
+        $this->domainEventCollector->collect(
+            new SnippetTranslationRemovedEvent(
+                $document,
+                $locale
+            )
+        );
+    }
+
+    public function handleCopyLocale(CopyLocaleEvent $event): void
+    {
+        $document = $event->getDocument();
+
+        if (!$document instanceof SnippetDocument) {
+            return;
+        }
+
+        $destDocument = $event->getDestDocument();
+
+        if (!$destDocument instanceof SnippetDocument) {
+            return;
+        }
+
+        $destLocale = $event->getDestLocale();
+        $sourceLocale = $event->getLocale();
+        $payload = $this->getPayloadFromSnippetDocument($destDocument);
+
+        $this->domainEventCollector->collect(
+            new SnippetTranslationCopiedEvent(
+                $destDocument,
+                $destLocale,
+                $sourceLocale,
+                $payload
+            )
+        );
+    }
+
+    public function handleCopy(CopyEvent $event): void
+    {
+        $document = $event->getDocument();
+
+        if (!$document instanceof SnippetDocument) {
+            return;
+        }
+
+        $this->eventsToBeDispatchedAfterFlush[] = [
+            'type' => SnippetCopiedEvent::class,
+            'options' => [
+                'snippetPath' => $event->getCopiedPath(),
+                'locale' => $document->getLocale(),
+                'sourceSnippetId' => $document->getUuid(),
+                'sourceSnippetTitle' => $document->getTitle(),
+            ],
+        ];
+    }
+
+    /**
+     * @return mixed[]
+     */
+    private function getPayloadFromSnippetDocument(SnippetDocument $snippetDocument): array
+    {
+        $data = $snippetDocument->getStructure()->toArray();
+
+        /** @var ExtensionContainer|mixed[] $extensionData */
+        $extensionData = $snippetDocument->getExtensionsData();
+
+        if ($extensionData instanceof ExtensionContainer) {
+            $extensionData = $extensionData->toArray();
+        }
+
+        $data['ext'] = $extensionData;
+
+        return $data;
+    }
+
+    /**
+     * @param NodeInterface<mixed> $node
+     *
+     * @see SecuritySubscriber::handlePersistCreate()
+     */
+    private function isNewNode(NodeInterface $node): bool
+    {
+        /** @var \Countable $properties */
+        $properties = $node->getProperties(
+            $this->propertyEncoder->encode(
+                'system_localized',
+                self::TITLE_FIELD,
```

Comment:
is there a reason for not using `StructureSubscriber::STRUCTURE_TYPE_FIELD` like for pages? going to trust you here 😇

Reply:
the structure field is not translated for the articles and the title is also a required field :)